docs/internals/contributing/writing-documentation.txt

```diff
@@ -341,6 +341,39 @@ documentation:
     Five
     ^^^^
 
+* Add headers for functions, classes, and methods in reference documentation:
+
+  - Each function, class, method, or attribute in reference documentation should
+    have a header using the appropriate heading level based on document hierarchy.
```

Comment:
```suggestion
    have a header using the appropriate heading level based on document hierarchy.
  - The header should be rendered as fixed width text by wrapping it with
    double backticks.
```

I think we should standardise on having code blocks/double backticks around headers of specific class, methods, functions/attributes and it would be good to explicitly mention this.